server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -96,6 +97,226 @@ public void teardown() {
     }
 
     @Test
+    public void setMailboxesCreateMailboxShouldSubscribeUser() throws Exception{
+        String requestBody =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"create\": {" +
+                "        \"create-id01\" : {" +
+                "          \"name\" : \"foo\"" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        given()
+            .header("Authorization", this.accessToken.serialize())
+            .body(requestBody)
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxesSet"))
+            .body(ARGUMENTS + ".created", hasKey("create-id01"));
+
+        assertThat(jmapServer.serverProbe().listSubscriptions(username)).containsOnly("foo");
```

Comment:
In fact, inbox is always default mailbox whatever it's subscribed or not. But yes, I will change to contains in case of it has some default mailboxes